packages/picker/src/Picker.ts

```diff
@@ -256,32 +251,27 @@ export class PickerBase extends SizedMixin(Focusable) {
         if (
             !this.popover ||
             !this.optionsMenu ||
-            this.optionsMenu.children.length === 0
+            !this.reparentableChildren ||
+            this.reparentableChildren.length === 0
         ) {
             this.menuStateResolver();
             return;
         }
 
-        this.placeholder = document.createComment(
-            'placeholder for optionsMenu'
+        this.restoreChildren = reparentChildren(
+            this.reparentableChildren,
+            this.optionsMenu
         );
 
         this.optionsMenu.selectable = true;
 
-        const parentElement =
-            this.optionsMenu.parentElement || this.optionsMenu.getRootNode();
-
-        if (parentElement) {
-            parentElement.replaceChild(this.placeholder, this.optionsMenu);
-        }
-
-        this.popover.append(this.optionsMenu);
         this.sizePopover(this.popover);
         const { popover } = this;
         this.closeOverlay = await Picker.openOverlay(this, 'inline', popover, {
             placement: this.placement,
             receivesFocus: 'auto',
         });
+        await this.manageSelection();
```

Comment:
Because we're now reparenting into the `sp-menu`, we need to manage selection for the menu to properly update. @Westbrook let me know if this approach makes sense. I'm not totally confident I'm covering everything yet.

Reply:
Yeah -- we'll want to do this in the next iteration. I can already see how #1189 will simplify a lot of the internal logic here.